imagepipeline/src/main/java/com/facebook/imagepipeline/producers/DecodeProducer.java

```diff
@@ -122,6 +122,9 @@ public void run(EncodedImage encodedImage, boolean isLast) {
               ImageRequest request = producerContext.getImageRequest();
               if (mDownsampleEnabledForNetwork ||
                   !UriUtil.isNetworkUri(request.getSourceUri())) {
+                if (!EncodedImage.isMetaDataAvailable(encodedImage)) {
```

Comment:
This is currently done in `AddImageTransformMetaDataProducer` which is included in the producer sequence for local images so shouldn't be needed here.